src/renderer/NumberInput.tsx

```diff
@@ -12,31 +12,40 @@ import ArrowButton from './ArrowButton';
 import './styles/NumberInput.scss';
 import { clamp } from './utils';
 
-type TNumberInput = number | '' | '-';
+type TNumberInput = string;
+type TInputType = 'int' | 'float';
 
 interface INumberInputProps {
   name: string;
-  value: TNumberInput;
+  value: number;
+  type: TInputType;
   min: number;
   max: number;
   isDisabled: boolean;
   showArrows: boolean;
+  floatPrecision?: number;
+  round?: boolean;
   showLabel: boolean;
   handleSubmit: (newValue: number) => Promise<void>;
 }
 
 const NumberInput = ({
   name,
   value,
+  type = 'int',
   min,
   max,
   isDisabled,
   showArrows,
+  floatPrecision = 0.1,
```

Comment:
Agreed upon by Frank and I to refactor this and the type to only be the `precision` which will represent the number of digits after the decimal point